### PR TITLE
Fat 731 c591 duplicate

### PR DESCRIPTION
**Pull request by TetianaParanich**

### CODE REVIEW

File: cypress/integration/checkin/checkin.basic.spec.js [hunk 1]
```diff
@@ -1,11 +1,13 @@
 import generateItemBarcode from '../../support/utils/generateItemBarcode';
 import checkoutActions from '../../support/fragments/checkout/checkout';
+import InventoryHoldings from '../../support/fragments/inventory/holdings/inventoryHoldings';
 
 describe('Check In', () => {
   let ITEM_BARCODE;
 
   beforeEach(() => {
     ITEM_BARCODE = generateItemBarcode();
```
Comment (MaksimTerekhov) [inline cypress/integration/checkin/checkin.basic.spec.js:9]: You can just assign value when you are creating variable

`let ITEM_BARCODE = generateItemBarcode();`

File: cypress/integration/checkout/checkout.basic.spec.js [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import generateItemBarcode from '../../support/utils/generateItemBarcode';
+import InventoryHoldings from '../../support/fragments/inventory/holdings/inventoryHoldings';
 
 describe('Check Out', () => {
   let ITEM_BARCODE;
 
   beforeEach(() => {
     ITEM_BARCODE = generateItemBarcode();
+    let source;
```
Comment (MaksimTerekhov) [inline cypress/integration/checkout/checkout.basic.spec.js:9]: the same

File: cypress/integration/inventory/item-locations.spec.js [hunk 3]
```diff
@@ -6,15 +6,16 @@ import InventoryInstanceEdit from '../../support/fragments/inventory/InventoryIn
...
 import devTeams from '../../support/dictionary/devTeams';
 
-
 const ITEM_BARCODE = `123${getRandomPostfix()}`;
 let userId = '';
 
-
 describe('ui-inventory: location', () => {
   before('create inventory instance', () => {
+    let source;
```
Comment (MaksimTerekhov) [inline cypress/integration/inventory/item-locations.spec.js:17]: the same

File: cypress/integration/checkout/checkout.basic.spec.js (hunk 2 above)
Comment (TetianaParanich) [reply cypress/integration/checkout/checkout.basic.spec.js:9]: Done

File: cypress/integration/checkout/checkout.basic.spec.js [hunk 4]
```diff
@@ -1,10 +1,12 @@
 import generateItemBarcode from '../../support/utils/generateItemBarcode';
+import InventoryHoldings from '../../support/fragments/inventory/holdings/inventoryHoldings';
 
 describe('Check Out', () => {
   let ITEM_BARCODE;
 
   beforeEach(() => {
     ITEM_BARCODE = generateItemBarcode();
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/integration/checkout/checkout.basic.spec.js:8]: Done

File: cypress/integration/circulation-log/search.spec.js [hunk 5]
```diff
@@ -3,18 +3,22 @@ import TestTypes from '../../support/dictionary/testTypes';
...
 import devTeams from '../../support/dictionary/devTeams';
+import UsersEditPage from '../../support/fragments/users/usersEditPage';
 
 const ITEM_BARCODE = `123${getRandomPostfix()}`;
 let userId = '';
 
 
 describe('ui-circulation-log', () => {
   before('create inventory instance', () => {
+    let source;
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/integration/circulation-log/search.spec.js:20]: Done

File: cypress/integration/inventory/filter-items-with-status.spec.js [hunk 6]
```diff
@@ -13,6 +14,8 @@ const title = `Filter items with status test ${Number(new Date())}`;
 
 describe('ui-inventory: items with status', () => {
   before('create inventory instance', () => {
+    let source;
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/integration/inventory/filter-items-with-status.spec.js:17]: Done

File: cypress/integration/inventory/item-locations.spec.js (hunk 3 above)
Comment (TetianaParanich) [reply cypress/integration/inventory/item-locations.spec.js:17]: Done

File: cypress/integration/checkout/checkout.multipiece.items.spec.js [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+    cy.wrap(testInstanceIds.holdingIds.forEach(holdingsId => {
+      cy.wrap(holdingsId.itemIds.forEach(itemId => {
+        cy.deleteItem(itemId);
+      })).then(() => {
+        cy.deleteHoldingRecord(holdingsId.id);
+      });
+    })).then(() => {
+      cy.deleteInstanceApi(testInstanceIds.instanceId);
+    });
+    // TODO delete service point and user
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/integration/checkout/checkout.multipiece.items.spec.js]: I added delete users and service point

File: cypress/integration/checkout/checkout.multipiece.items.spec.js [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+    SwitchServicePoint.changeServicePointPreference(user.username);
+    cy.deleteServicePoint(servicePoint.body.id);
+    Users.deleteViaApi(user.userId);
+  });
+
+  it('C591 Check out: multipiece items', { tags: [TestTypes.smoke] }, () => {
+    CheckOutActions.checkIsInterfacesOpened();
+    CheckOutActions.checkOutItem(userBarcode, testItems[0].barcode);
+    CheckOutActions.checkPatronInformation(user.username, userBarcode);
+    CheckOutActions.checkConfirmMultipieceCheckOut();
```
Comment (dmitriyfolio) [inline cypress/integration/checkout/checkout.multipiece.items.spec.js:118]: add isPresented into checkConfirmMultipieceCheckOut  to check in one place availability or absence

File: cypress/support/fragments/checkout/modals/multipieceCheckOut.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+      cy.expect(descriptionOfPieces.has({ value: description }));
+    }
+    if (missingItemPieces === missingDescription && missingItemPieces === '-') {
+      cy.expect(numberOfMissingPiecesKeyValue.absent());
+      cy.expect(descriptionOfMissingPiecesKeyValue.absent());
+    } else {
+      cy.expect(numberOfMissingPiecesKeyValue.has({ value: missingItemPieces.toString() }));
+      cy.expect(descriptionOfMissingPiecesKeyValue.has({ value: missingDescription }));
+    }
+    cy.expect(checkOutButton.exists());
```
Comment (dmitriyfolio) [inline cypress/support/fragments/checkout/modals/multipieceCheckOut.js:33]: improve checking based on "Two active buttons: Cancel or Check out. Check out has primary styling."

**Review by MaksimTerekhov — APPROVED** [review]

File: cypress/integration/checkout/checkout.multipiece.items.spec.js (hunk 8 above)
Comment (TetianaParanich) [reply cypress/integration/checkout/checkout.multipiece.items.spec.js]: Done

File: cypress/support/fragments/checkout/modals/multipieceCheckOut.js (hunk 9 above)
Comment (TetianaParanich) [reply cypress/support/fragments/checkout/modals/multipieceCheckOut.js:33]: Done

File: cypress/integration/loans/loan-dates.spec.js [hunk 10]
```diff
@@ -9,7 +9,11 @@ import ChangeDueDateForm from '../../support/fragments/loans/changeDueDateForm';
 import CheckOutActions from '../../support/fragments/check-out-actions/check-out-actions';
 import DateTools from '../../support/utils/dateTools';
 import NewRequest from '../../support/fragments/requests/newRequest';
-import checkinActions from '../../support/fragments/check-in-actions/checkInActions';
+import CheckinActions from '../../support/fragments/check-in-actions/checkInActions';
+import ConfirmMultiplePiecesItemCheckOut from '../../support/fragments/checkout/confirmMultipieceCheckOutModal';
```
Comment (dmitriyfolio) [inline cypress/integration/loans/loan-dates.spec.js:13]: fix linter issue

**Review by dmitriyfolio — APPROVED** [review]

**Pull request merged**